errtracker/errtracker.go

```diff
@@ -59,16 +66,35 @@ func distroRelease() string {
 	return fmt.Sprintf("%s %s", ID, release.ReleaseInfo.VersionID)
 }
 
+func readMachineID() ([]byte, error) {
+	var machineID []byte
+	var err error
+	for _, id := range machineIDs {
+		machineID, err = ioutil.ReadFile(id)
+		if err == nil {
+			break
```

Comment:
You can simplify this code by returning here: `return bytes.TrimSpace(machineID), nil`

Then you can write after the loop: `return nil, fmt.Errorf("cannot report: no suitable machine id file found")` and avoid the len(machineID) check altogether.